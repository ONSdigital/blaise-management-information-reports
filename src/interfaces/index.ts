interface InterviewerCallHistoryReportData {
    questionnaire_name: string
    serial_number: string
    call_start_time: string
    dial_secs: string
    call_result: string
}

interface AppointmentResourcePlanningReportData {
    questionnaire_name: string
    appointment_time: string
    appointment_language: string
    total: number
}

export type {InterviewerCallHistoryReportData, AppointmentResourcePlanningReportData};
