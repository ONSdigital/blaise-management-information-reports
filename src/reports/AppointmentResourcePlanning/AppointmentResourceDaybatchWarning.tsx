import { ONSPanel } from "blaise-design-system-react-components";
import React, { ReactElement } from "react";

const AppointmentResourceDaybatchWarning = (): ReactElement => (
    <>
        <ONSPanel>
            <p>
                Run a Daybatch first to obtain the most accurate results.
            </p>
            <p>
                Appointments that have already been attempted will not be displayed.
            </p>
        </ONSPanel>
    </>
);

export default AppointmentResourceDaybatchWarning;
