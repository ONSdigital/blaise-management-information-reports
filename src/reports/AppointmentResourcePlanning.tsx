import React, {ReactElement, useState} from "react";
import {ONSButton, ONSPanel} from "blaise-design-system-react-components";
import Form from "../components/Form";
import {getAppointmentResourcePlanningReport} from "../utilities/HTTP";
import {AppointmentResourcePlanningReportData} from "../interfaces";
import {ErrorBoundary} from "../components/ErrorHandling/ErrorBoundary";
import {ONSDateInput} from "../components/ONSDesignSystem/ONSDateInput";
import dateFormatter from "dayjs";
import utc from "dayjs/plugin/utc";
import timezone from "dayjs/plugin/timezone";
import Breadcrumbs from "../components/Breadcrumbs";

dateFormatter.extend(utc);
dateFormatter.extend(timezone);

function AppointmentResourcePlanning(): ReactElement {
    const [buttonLoading, setButtonLoading] = useState<boolean>(false);
    const [date, setDate] = useState<Date>(new Date());
    const [message, setMessage] = useState<string>("");
    const [messageNoData, setMessageNoData] = useState<string>("");
    const [reportData, setReportData] = useState<AppointmentResourcePlanningReportData[]>([]);

    async function runAppointmentResourcePlanningReport(formData: any) {
        setMessageNoData("");
        setReportData([]);
        setButtonLoading(true);
        console.log(formData);
        formData.date = date;

        const [success, data] = await getAppointmentResourcePlanningReport(formData);
        setButtonLoading(false);

        if (!success) {
            setMessage("Error running report");
            return;
        }

        if (data.length == 0) {
            setMessageNoData("No data found for parameters given.");
            return;
        }

        console.log(data);
        setReportData(data);
    }

    return (
        <>
            <Breadcrumbs BreadcrumbList={[{link: "/", title: "Back"}]}/>
            <main id="main-content" className="page__main u-mt-s">
                <h1 className="u-mb-m">Run appointment resource planning report</h1>
                <ONSPanel hidden={(message === "")} status="error">
                    {message}
                </ONSPanel>
                <Form onSubmit={(data) => runAppointmentResourcePlanningReport(data)}>
                    <ONSDateInput
                        label={"Date"}
                        date={date}
                        id={"date"}
                        onChange={(date) => setDate(date)}
                    />
                    <br/>
                    <br/>
                    <ONSButton
                        testid={"submit-Form"}
                        label={"Run"}
                        primary={true}
                        loading={buttonLoading}
                        submit={true}/>
                </Form>
                <br/>

                <ErrorBoundary errorMessageText={"Failed to load"}>
                    {
                        reportData && reportData.length > 0
                            ?
                            <table id="report-table" className="table u-mt-s">
                                <thead className="table__head u-mt-m">
                                <tr className="table__row">
                                    <th scope="col" className="table__header ">
                                        <span>Questionnaire</span>
                                    </th>
                                    <th scope="col" className="table__header ">
                                        <span>Appointment Time</span>
                                    </th>
                                    <th scope="col" className="table__header ">
                                        <span>Appointment Language</span>
                                    </th>
                                    <th scope="col" className="table__header ">
                                        <span>Call Result</span>
                                    </th>
                                    <th scope="col" className="table__header ">
                                        <span>Total</span>
                                    </th>
                                </tr>
                                </thead>
                                <tbody className="table__body">
                                {
                                    reportData.map((data: AppointmentResourcePlanningReportData) => {
                                        return (
                                            <tr className="table__row" key={data.questionnaire_name}
                                                data-testid={"report-table-row"}>
                                                <td className="table__cell ">
                                                    {data.questionnaire_name}
                                                </td>
                                                <td className="table__cell ">
                                                    {data.appointment_time}
                                                </td>
                                                <td className="table__cell ">
                                                    {data.appointment_language}
                                                </td>
                                                <td className="table__cell ">
                                                    {data.dial_result}
                                                </td>
                                                <td className="table__cell ">
                                                    {data.total}
                                                </td>
                                            </tr>
                                        );
                                    })
                                }
                                </tbody>
                            </table>
                            :
                            <ONSPanel hidden={messageNoData === "" && true}>{messageNoData}</ONSPanel>
                    }
                    <br/>
                </ErrorBoundary>
            </main>
        </>
    );
}

export default AppointmentResourcePlanning;
