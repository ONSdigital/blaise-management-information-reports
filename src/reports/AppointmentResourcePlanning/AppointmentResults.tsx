import { ErrorBoundary, ONSPanel } from "blaise-design-system-react-components";
import React, { ReactElement } from "react";
import { AppointmentResourcePlanningReportData } from "../../interfaces";

interface Props {
    reportData: AppointmentResourcePlanningReportData[],
    messageNoData: string
}

export default function AppointmentResults({
    messageNoData,
    reportData,
}: Props): ReactElement {
    return (
        <ErrorBoundary errorMessageText="Failed to load">
            {
                reportData && reportData.length > 0
                    ? (
                        <table id="report-table" className="ons-table ons-elementToFadeIn ons-u-mt-s">
                            <thead className="ons-table__head u-mt-m">
                                <tr className="ons-table__row">
                                    <th scope="col" className="ons-table__header ">
                                        <span>Questionnaire</span>
                                    </th>
                                    <th scope="col" className="ons-table__header ">
                                        <span>Appointment Time</span>
                                    </th>
                                    <th scope="col" className="ons-table__header ">
                                        <span>Appointment Language</span>
                                    </th>
                                    <th scope="col" className="ons-table__header ">
                                        <span>Case Reference</span>
                                    </th>
                                    <th scope="col" className="ons-table__header ">
                                        <span>Telephone Number</span>
                                    </th>
                                    <th scope="col" className="ons-table__header ">
                                        <span>Respondent Name</span>
                                    </th>
                                </tr>
                            </thead>
                            <tbody className="ons-table__body">
                                {
                                    reportData.map((data: AppointmentResourcePlanningReportData) => (
                                        <tr
                                            className="ons-table__row"
                                            key={`${data.questionnaire_name}-${data.appointment_time}-${data.appointment_language}-${data.case_reference}`}
                                            data-testid="report-table-row"
                                        >
                                            <td className="ons-table__cell ">
                                                {data.questionnaire_name}
                                            </td>
                                            <td className="ons-table__cell ">
                                                {data.appointment_time}
                                            </td>
                                            <td className="ons-table__cell ">
                                                {data.appointment_language}
                                            </td>
                                            <td className="ons-table__cell ">
                                                {data.case_reference}
                                            </td>
                                            <td className="ons-table__cell ">
                                                {data.telephone_number?.replace(/'/g, "")}
                                            </td>
                                            <td className="ons-table__cell ">
                                                {data.respondent_name?.replace(/'/g, "")}
                                            </td>
                                        </tr>
                                    ))
                                }
                            </tbody>
                        </table>
                    )
                    : <ONSPanel hidden={messageNoData === "" && true}>{messageNoData}</ONSPanel>
            }
            <br />
        </ErrorBoundary>
    );
}
