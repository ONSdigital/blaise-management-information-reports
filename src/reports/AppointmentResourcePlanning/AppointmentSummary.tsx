import React, { ReactElement } from "react";
import { ONSPanel } from "blaise-design-system-react-components";
import { AppointmentResourcePlanningSummaryReportData } from "../../interfaces";

interface Props {
    data: AppointmentResourcePlanningSummaryReportData[];
    failed: boolean;
}

const AppointmentSummary = ({ data, failed }: Props): ReactElement => {
    if (failed) {
        return <ONSPanel status="error"><p>Failed to get appointment language summary</p></ONSPanel>;
    }

    if (data.length === 0) {
        return <></>;
    }

    return (
        <div>
            <div className="summary u-mb-m elementToFadeIn u-mt-m">
                <div className="summary__group">
                    <h2 className="summary__group-title">Appointment language summary</h2>
                    <table className="summary__items u-mt-s">
                        <thead className="u-vh">
                            <tr>
                                <th>Language</th>
                                <th>Total appointments</th>
                            </tr>
                        </thead>
                        {
                            data.map(({ language, total }: AppointmentResourcePlanningSummaryReportData) => (
                                <tbody className="summary__item" key={language}>
                                    <tr
                                        className="summary__row summary__row--has-values"
                                        data-testid="summary-table-row"
                                    >
                                        <td className="summary__item-title">
                                            <div className="summary__item--text">
                                                {language}
                                            </div>
                                        </td>
                                        <td className="summary__values">
                                            {total}
                                        </td>
                                    </tr>
                                </tbody>
                            ))
                        }
                    </table>
                </div>
            </div>
        </div>
    );
};

export default AppointmentSummary;
