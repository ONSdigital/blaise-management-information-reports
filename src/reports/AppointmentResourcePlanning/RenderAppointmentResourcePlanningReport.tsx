import React, { ReactElement, useEffect, useState } from "react";
import { CSVLink } from "react-csv";
import Breadcrumbs from "../../components/Breadcrumbs";
import { AppointmentResourcePlanningReportData, AppointmentResourcePlanningSummaryReportData } from "../../interfaces";
import { getAppointmentResourcePlanningReport, getAppointmentResourcePlanningSummaryReport } from "../../utilities/HTTP";
import AppointmentResourceDaybatchWarning from "./AppointmentResourceDaybatchWarning";
import ReportErrorPanel from "../../components/ReportErrorPanel";
import AppointmentResults from "./AppointmentResults";
import AppointmentSummary from "./AppointmentSummary";
import { formatDate } from "../../utilities/DateFormatter";

interface RenderAppointmentResourcePlanningReportPageProps {
    reportDate: Date;
    surveyTla: string;
    questionnaires: string[];
    navigateBack: () => void;
    navigateBackTwoSteps: () => void;
}

function formatList(listOfQuestionnaires: string[]): string {
    if (listOfQuestionnaires.length === 1) return listOfQuestionnaires[0];
    const firsts = listOfQuestionnaires.slice(0, listOfQuestionnaires.length - 1);
    const last = listOfQuestionnaires[listOfQuestionnaires.length - 1];
    return `${firsts.join(", ")} and ${last}`;
}

function RenderAppointmentResourcePlanningReport(props: RenderAppointmentResourcePlanningReportPageProps): ReactElement {
    const [reportFailed, setReportFailed] = useState<boolean>(false);
    const [reportData, setReportData] = useState<AppointmentResourcePlanningReportData[]>([]);
    const [messageNoData, setMessageNoData] = useState<string>("");
    const [summaryFailed, setSummaryFailed] = useState<boolean>(false);
    const [summaryData, setSummaryData] = useState<AppointmentResourcePlanningSummaryReportData[]>([]);

    const {
        reportDate,
        surveyTla,
        questionnaires,
        navigateBack,
        navigateBackTwoSteps,
    } = props;

    const reportExportHeaders = [
        { label: "Questionnaire", key: "questionnaire_name" },
        { label: "Appointment Time", key: "appointment_time" },
        { label: "Appointment Language", key: "appointment_language" },
        { label: "Case Reference", key: "case_reference" },
        { label: "Telephone Number", key: "telephone_number" },
        { label: "Respondent Name", key: "respondent_name" },
    ];

    useEffect(() => {
        async function runAppointmentResourcePlanningReport(): Promise<void> {
            setMessageNoData("");
            setReportFailed(false);
            setReportData([]);

            let planningReport: AppointmentResourcePlanningReportData[];
            try {
                planningReport = await getAppointmentResourcePlanningReport(reportDate, surveyTla, questionnaires);
            } catch {
                setReportFailed(true);
                return;
            } finally {
                // setSubmitting(false);
            }

            if (planningReport.length === 0) {
                setMessageNoData("No data found for parameters given.");
                return;
            }

            console.log(planningReport);
            setReportData(planningReport);
        }

        runAppointmentResourcePlanningReport();
    }, [questionnaires, reportDate, surveyTla]);

    useEffect(() => {
        async function runAppointmentSummary(): Promise<void> {
            setSummaryData([]);
            setSummaryFailed(false);
            getAppointmentResourcePlanningSummaryReport(reportDate, surveyTla, questionnaires)
                .then((summaryReport: AppointmentResourcePlanningSummaryReportData[]) => {
                    console.log(summaryReport);
                    setSummaryData(summaryReport);
                }).catch(() => {
                    setSummaryFailed(true);
                });
        }

        runAppointmentSummary();
    }, [questionnaires, reportDate, surveyTla]);

    return (
        <>
            <Breadcrumbs BreadcrumbList={[{ link: "/", title: "Reports" }, {
                link: "#",
                onClickFunction: navigateBackTwoSteps,
                title: "Appointment details",
            }, { link: "#", onClickFunction: navigateBack, title: "Questionnaires" }]}
            />
            <main id="main-content" className="page__main u-mt-s">

                <h1 className="u-mb-m">
                    Appointment Resource Planning Report
                </h1>
                <h3 className="u-mb-m">
                    Date: {formatDate(reportDate)}<br />
                    Questionnaire{questionnaires.length > 1 ? ("s:") : ":"} {formatList(questionnaires)}{" "}
                </h3>
                <AppointmentResourceDaybatchWarning />

                <ReportErrorPanel error={reportFailed} />

                <AppointmentSummary data={summaryData} failed={summaryFailed} />
                <div className=" u-mt-m">
                    <CSVLink
                        hidden={reportData === null || reportData.length === 0}
                        data={reportData}
                        headers={reportExportHeaders}
                        target="_blank"
                        filename={`appointment-resource-planning-report-${reportDate}.csv`}
                    >
                        Export report as Comma-Separated Values (CSV) file
                    </CSVLink>
                </div>
                <AppointmentResults reportData={reportData} messageNoData={messageNoData} />
            </main>
        </>
    );
}

export default RenderAppointmentResourcePlanningReport;
