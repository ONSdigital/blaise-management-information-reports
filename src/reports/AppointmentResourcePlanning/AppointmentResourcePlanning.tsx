import React, { ReactElement, useState } from "react";
import AppointmentFilter from "../filters/AppointmentFilter";
import AppointmentQuestionnaireFilter from "../filters/AppointmentQuestionnaireFilter";
import RenderAppointmentResourcePlanningReport from "./RenderAppointmentResourcePlanningReport";

enum Step {
    AppointmentFilter,
    QuestionnaireFilter,
    RenderReport,
}

function AppointmentResourcePlanning(): ReactElement {
    const [activeStep, setActiveStep] = useState<Step>(Step.AppointmentFilter);
    const [reportDate, setReportDate] = useState<Date>(new Date());
    const [surveyTla, setSurveyTla] = useState<string>("");
    const [questionnaires, setQuestionnaires] = useState<string[]>([]);

    function _renderStepContent(step: number) {
        switch (step) {
        case Step.AppointmentFilter:
            return (<AppointmentFilter title="appointment resource planning"
                reportDate={reportDate} setReportDate={setReportDate}
                surveyTla={surveyTla} setSurveyTla={setSurveyTla}
                submitFunction={_handleSubmit}/>);
        case Step.QuestionnaireFilter:
            return (<AppointmentQuestionnaireFilter reportDate={reportDate}
                surveyTla={surveyTla}
                questionnaires={questionnaires} setQuestionnaires={setQuestionnaires}
                submitFunction={_handleSubmit}
                navigateBack={_navigateBack}/>);
        case Step.RenderReport:
            console.log(`Steps questionnaires ${questionnaires}`);
            return (<RenderAppointmentResourcePlanningReport reportDate={reportDate}
                surveyTla={surveyTla}
                questionnaires={questionnaires}
                navigateBack={_navigateBack}
                navigateBackTwoSteps={_navigateBackTwoSteps}/>);
        }

    }

    async function _handleSubmit() {
        switch (activeStep) {
        case Step.AppointmentFilter:
            setActiveStep(Step.QuestionnaireFilter);
            break;
        case Step.QuestionnaireFilter:
            setActiveStep(Step.RenderReport);
            break;
        default:
            setActiveStep(Step.AppointmentFilter);
        }
    }

    function _navigateBack() {
        setActiveStep(activeStep - 1);
    }

    function _navigateBackTwoSteps() {
        setActiveStep(activeStep - 2);
    }

    return (
        <div>
            <div className="u-mt-m">
                {_renderStepContent(activeStep)}
            </div>
        </div>
    );
}

export default AppointmentResourcePlanning;
