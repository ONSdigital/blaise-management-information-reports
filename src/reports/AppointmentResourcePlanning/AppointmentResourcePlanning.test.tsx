/**
 * @jest-environment jsdom
 */

import { AppointmentResourcePlanningReportData } from "../../interfaces";
import "@testing-library/jest-dom";
import flushPromises from "../../tests/utilities";
import { createMemoryHistory } from "history";
import { cleanup, render } from "@testing-library/react";
import { Router } from "react-router";
import AppointmentResourcePlanning from "./AppointmentResourcePlanning";
import { act } from "react-dom/test-utils";
import { screen } from "@testing-library/dom";
import React from "react";
import MockDate from "mockdate";
import MockAdapter from "axios-mock-adapter";
import axios from "axios";
import userEvent from "@testing-library/user-event";


const mockAdapter = new MockAdapter(axios);

const ReportSummary = [
    { language: "English", total: 1 },
    { language: "Welsh", total: 1 },
    { language: "Other", total: 1 },
];

const reportDataReturned: AppointmentResourcePlanningReportData[] = [
    {
        questionnaire_name: "LMS2101_AA1",
        appointment_time: "10:00",
        appointment_language: "English",
        total: 42
    },
    {
        questionnaire_name: "LMS2101_BB1",
        appointment_time: "12:30",
        appointment_language: "Welsh",
        total: 1908
    },
    {
        questionnaire_name: "LMS2101_CC1",
        appointment_time: "15:15",
        appointment_language: "Other",
        total: 408
    }
];

const questionnairesReturned = ["LMS2101_AA1", "LMS2101_BB1", "LMS2101_CC1"]

const christmasEve97 = "1997-12-24";


describe("appointment resource planning report without data", () => {
    beforeEach(() => {
        MockDate.set(new Date(christmasEve97));
        mockAdapter.onPost("/api/reports/appointment-resource-planning").reply(200, "");
        mockAdapter.onPost("/api/reports/appointment-resource-planning-summary").reply(200, []);
        mockAdapter.onPost("/api/appointments/questionnaires").reply(200, []);
    });

    afterEach(() => {
        MockDate.reset();
        mockAdapter.reset();
    });

    it("matches snapshot", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <AppointmentResourcePlanning />
            </Router>
        );
        await act(async () => {
            await flushPromises();
        });

        expect(await wrapper).toMatchSnapshot();
    });
    it("renders correctly", async () => {
        const history = createMemoryHistory();
        await act(async () => {
            render(
                <Router history={history}>
                    <AppointmentResourcePlanning />
                </Router>
            );
        });
        expect(screen.queryByText("Run appointment resource planning report")).toBeVisible();
        expect(screen.queryByText("Run a Daybatch first to obtain the most accurate results.")).toBeVisible();
        expect(screen.queryByText("Appointments that have already been attempted will not be displayed.")).toBeVisible();
        
        expect(screen.queryByText("Select survey")).toBeVisible();
        expect(screen.queryByText("Show all surveys")).toBeVisible();
        expect(screen.queryByText("LMS")).toBeVisible();
        expect(screen.queryByText("Labour Market Survey")).toBeVisible();
        expect(screen.queryByText("OPN")).toBeVisible();
        expect(screen.queryByText("Opinions and Lifestyle Survey")).toBeVisible();

        expect(screen.queryByText("Date")).toBeVisible();

        userEvent.type(screen.getByLabelText(/Date/i), "2021-01-01");
        userEvent.click(screen.getByTestId(/submit-button/i));

        await act(async () => {
            await flushPromises();
        });

        expect(await screen.queryByText("No questionnaires found for given parameters.")).toBeVisible();
    });
});
