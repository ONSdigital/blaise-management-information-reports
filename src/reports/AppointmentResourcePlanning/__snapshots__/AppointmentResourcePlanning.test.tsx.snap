// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appointment resource planning report without data matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ons-u-mt-m"
        >
          <div>
            <nav
              aria-label="Breadcrumb"
              class="ons-breadcrumb"
            >
              <ol
                class="ons-breadcrumb__items ons-u-fs-s"
              >
                <li
                  class="ons-breadcrumb__item"
                  id="breadcrumb-0"
                >
                  <a
                    class="ons-breadcrumb__link"
                    href="/"
                  >
                    Reports
                  </a>
                  <svg
                    class="ons-svg-icon"
                    focusable="false"
                    viewBox="0 0 8 13"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                      transform="translate(-5.02 -1.59)"
                    />
                  </svg>
                </li>
              </ol>
            </nav>
            <main
              class="ons-page__main ons-u-mt-s"
              id="main-content"
            >
              <h1
                class="ons-u-mb-m"
              >
                Run 
                appointment resource planning
                 report
              </h1>
              <div
                class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
              >
                <div
                  class="ons-panel__body "
                >
                  <p>
                    Run a Daybatch first to obtain the most accurate results.
                  </p>
                  <p>
                    Appointments that have already been attempted will not be displayed.
                  </p>
                </div>
              </div>
              <br />
              <form
                action="#"
              >
                <fieldset
                  class="ons-fieldset"
                >
                  <legend
                    class="ons-fieldset__legend"
                  >
                    Select survey
                  </legend>
                  <div
                    class="ons-radios__items"
                    id="Survey TLA"
                  >
                    <p
                      class="ons-radios__item"
                    >
                      <span
                        class="ons-radio"
                      >
                        <input
                          checked=""
                          class="ons-radio__input js-radio"
                          id="all"
                          initial_value=""
                          name="Survey TLA"
                          type="radio"
                          value=""
                        />
                        <label
                          class="ons-radio__label "
                          for="all"
                          id="all-label"
                        >
                          Show all surveys
                        </label>
                      </span>
                    </p>
                    <br />
                    <p
                      class="ons-radios__item"
                    >
                      <span
                        class="ons-radio"
                      >
                        <input
                          class="ons-radio__input js-radio"
                          id="lms"
                          initial_value=""
                          name="Survey TLA"
                          type="radio"
                          value="lms"
                        />
                        <label
                          class="ons-radio__label ons-label--with-description"
                          for="lms"
                          id="lms-label"
                        >
                          LMS
                          <span
                            class="ons-label__description ons-radio__label--with-description"
                            id="white-label-description-hint"
                          >
                            Labour Market Survey
                          </span>
                        </label>
                      </span>
                    </p>
                    <br />
                    <p
                      class="ons-radios__item"
                    >
                      <span
                        class="ons-radio"
                      >
                        <input
                          class="ons-radio__input js-radio"
                          id="opn"
                          initial_value=""
                          name="Survey TLA"
                          type="radio"
                          value="opn"
                        />
                        <label
                          class="ons-radio__label ons-label--with-description"
                          for="opn"
                          id="opn-label"
                        >
                          OPN
                          <span
                            class="ons-label__description ons-radio__label--with-description"
                            id="white-label-description-hint"
                          >
                            Opinions and Lifestyle Survey
                          </span>
                        </label>
                      </span>
                    </p>
                    <br />
                  </div>
                </fieldset>
                <div
                  class="ons-field"
                >
                  <label
                    class="ons-label "
                    for="Date"
                  >
                    Date
                  </label>
                  <input
                    class="ons-input ons-input--text ons-input-type__input "
                    id="Date"
                    initial_value="1997-12-24"
                    name="Date"
                    type="date"
                    value="1997-12-24"
                  />
                </div>
                <br />
                <button
                  class="ons-btn "
                  data-testid="submit-button"
                  type="submit"
                >
                  <span
                    class="ons-btn__inner"
                  >
                    Next
                  </span>
                </button>
              </form>
            </main>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ons-u-mt-m"
      >
        <div>
          <nav
            aria-label="Breadcrumb"
            class="ons-breadcrumb"
          >
            <ol
              class="ons-breadcrumb__items ons-u-fs-s"
            >
              <li
                class="ons-breadcrumb__item"
                id="breadcrumb-0"
              >
                <a
                  class="ons-breadcrumb__link"
                  href="/"
                >
                  Reports
                </a>
                <svg
                  class="ons-svg-icon"
                  focusable="false"
                  viewBox="0 0 8 13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                    transform="translate(-5.02 -1.59)"
                  />
                </svg>
              </li>
            </ol>
          </nav>
          <main
            class="ons-page__main ons-u-mt-s"
            id="main-content"
          >
            <h1
              class="ons-u-mb-m"
            >
              Run 
              appointment resource planning
               report
            </h1>
            <div
              class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
            >
              <div
                class="ons-panel__body "
              >
                <p>
                  Run a Daybatch first to obtain the most accurate results.
                </p>
                <p>
                  Appointments that have already been attempted will not be displayed.
                </p>
              </div>
            </div>
            <br />
            <form
              action="#"
            >
              <fieldset
                class="ons-fieldset"
              >
                <legend
                  class="ons-fieldset__legend"
                >
                  Select survey
                </legend>
                <div
                  class="ons-radios__items"
                  id="Survey TLA"
                >
                  <p
                    class="ons-radios__item"
                  >
                    <span
                      class="ons-radio"
                    >
                      <input
                        checked=""
                        class="ons-radio__input js-radio"
                        id="all"
                        initial_value=""
                        name="Survey TLA"
                        type="radio"
                        value=""
                      />
                      <label
                        class="ons-radio__label "
                        for="all"
                        id="all-label"
                      >
                        Show all surveys
                      </label>
                    </span>
                  </p>
                  <br />
                  <p
                    class="ons-radios__item"
                  >
                    <span
                      class="ons-radio"
                    >
                      <input
                        class="ons-radio__input js-radio"
                        id="lms"
                        initial_value=""
                        name="Survey TLA"
                        type="radio"
                        value="lms"
                      />
                      <label
                        class="ons-radio__label ons-label--with-description"
                        for="lms"
                        id="lms-label"
                      >
                        LMS
                        <span
                          class="ons-label__description ons-radio__label--with-description"
                          id="white-label-description-hint"
                        >
                          Labour Market Survey
                        </span>
                      </label>
                    </span>
                  </p>
                  <br />
                  <p
                    class="ons-radios__item"
                  >
                    <span
                      class="ons-radio"
                    >
                      <input
                        class="ons-radio__input js-radio"
                        id="opn"
                        initial_value=""
                        name="Survey TLA"
                        type="radio"
                        value="opn"
                      />
                      <label
                        class="ons-radio__label ons-label--with-description"
                        for="opn"
                        id="opn-label"
                      >
                        OPN
                        <span
                          class="ons-label__description ons-radio__label--with-description"
                          id="white-label-description-hint"
                        >
                          Opinions and Lifestyle Survey
                        </span>
                      </label>
                    </span>
                  </p>
                  <br />
                </div>
              </fieldset>
              <div
                class="ons-field"
              >
                <label
                  class="ons-label "
                  for="Date"
                >
                  Date
                </label>
                <input
                  class="ons-input ons-input--text ons-input-type__input "
                  id="Date"
                  initial_value="1997-12-24"
                  name="Date"
                  type="date"
                  value="1997-12-24"
                />
              </div>
              <br />
              <button
                class="ons-btn "
                data-testid="submit-button"
                type="submit"
              >
                <span
                  class="ons-btn__inner"
                >
                  Next
                </span>
              </button>
            </form>
          </main>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
